cmd/agent/common/autodiscovery.go

```diff
@@ -46,6 +46,8 @@ func setupAutoDiscovery(confSearchPaths []string, metaScheduler *scheduler.MetaS
 		time.Duration(config.Datadog.GetInt("autoconf_config_files_poll_interval"))*time.Second,
 	)
 
+	ad.AddConfigProvider(providers.NewTelemetryProvider(), false, 0)
```

Comment:
How can we disable this the check instance? 
I see two usecases:
* when the agent run as cluster-check-runner in kubernetes, we might want to use another configuration that the one hardcoded in the new TelemetryProvider. (change the metric namespace)
* Internally if we want to get more telemetry metrics, we would like to change the list of metrics that are scrapped.